subsys/spm/secure_services.c

```diff
@@ -41,6 +42,10 @@
 #include <mbedtls/entropy_poll.h>
 #endif /* CONFIG_SPM_SERVICE_RNG */
 
+static bool ptr_in_secure_ram(intptr_t ptr)
+{
+	return ptr < (CONFIG_SRAM_BASE_ADDRESS + NON_SECURE_RAM_OFFSET);
```

Comment:
Would it be possible, here to use the ARMv8-M intrisics to do this? See if you can use the Test Target for this, to make this code portable.

Reply:
Now I recall, I wrote an API for this, some years ago:
`int arm_cmse_addr_is_secure(uint32_t addr)
`
can you see if this works for you?

Reply:
Will try!

Reply:
Well that makes things a lot cleaner, thanks for the tip!